### DISCUSSION TITLE
53.4 Tasks

### DISCUSSION THREAD

**Discussion by njr0** [google/comprehensive-rust#1316] [General]
This is clearly framed as something that needs the Speaker to have some context and a "prepared `main.rs`".

That's fine if you only want the course (or at least, this section of the course) to be usable in a classroom context, but is not great for folks trying to do it themselves.

I made it run OK (by just adding `tokio = { version = "1.32.0", features = ["full"] }` as a dependency in `Cargo.toml`, and I could  get it to say `connection from 127.0.0.1:64382` etc. by using `curl`, but I guess I'd have to write code to connect to the socket and receive the question and send a reply and stuff. Presumably, that's what's in the "prepared `main.rs`.

I guess I think it would be helpful to include the prepared main (and maybe notes on what you need in `Cargo.toml`) in the speaker notes.

**Comment by mgeisler** [comment]
@sakex, @rbehjati, and @djmitche: can we make this more clear in the course?

**Reply by rbehjati** [reply]
Which part of the content is this about? @njr0 Can you add a pointer to the source?

But anyway, this sounds like something that should be explained in the course content.

**Reply by njr0** [reply]
@rbehjati Sorry, it's section 53.4, entitled Tasks.  (That's what I meant by 53.4 Tasks as the Discussion name.) It's here:

https://github.com/google/comprehensive-rust/blob/main/src/async/tasks.md

**Reply by djmitche** [reply]
The "prepared `main.rs`" refers to https://github.com/google/comprehensive-rust/blob/main/src/running-the-course/course-structure.md#concurrency-in-rust. Maybe just adding a link would be sufficient?

As for connecting, `telnet` or `nc` are good tools for that. Adding a note to that effect would be good.

**Reply by djmitche** [reply]
#1330 for the "prepared `main.rs`" bit.

#1349 for the suggestion to use `nc` or `telnet`.
